File: PeriodisationProgramApp.Domain/Entities/MuscleGroup.cs
using PeriodisationProgramApp.Domain.Enums;

namespace PeriodisationProgramApp.Domain.Entities
{
    public class MuscleGroup : BaseEntity
    {
        public MuscleGroupType Type { get; set; }

        public int MaintenanceVolume { get; set; }

        public int MinimumEffectiveVolume { get; set; }

        public int MaximumRecoverableVolume { get; set; }

        public int MaximumRecoverableVolumeMultiplicator { get; set; }

        public int AverageRecoveryTime { get; set; }

        public List<MuscleGroupUserData> MuscleGroupUsersData { get; set; } = new();

        public void Update(MuscleGroup other)
        {
            Type = other.Type;
            MaintenanceVolume = other.MaintenanceVolume;
            MinimumEffectiveVolume = other.MinimumEffectiveVolume;
            MaximumRecoverableVolume = other.MaximumRecoverableVolume;
            MaximumRecoverableVolumeMultiplicator = other.MaximumRecoverableVolumeMultiplicator;
            AverageRecoveryTime = other.AverageRecoveryTime;
        }
    }
}

File: PeriodisationProgramApp.Web/src/types/enitities/MuscleGroup.ts
import { MuscleGroupType } from "../../enums/MuscleGroupType";
import { MuscleGroupUserData } from "../MuscleGroupUserData";
import { BaseEntity } from "./BaseEntity";

export type MuscleGroup = {
  type: MuscleGroupType;
  maintenanceVolume: number;
  minimumEffectiveVolume: number;
  maximumRecoverableVolume: number;
  maximumRecoverableVolumeMultiplicator: number;
  averageRecoveryTime: number;
  muscleGroupUserData: MuscleGroupUserData | null;
} & BaseEntity;
